dashboard/src/shared/utils.ts

```diff
@@ -33,3 +36,28 @@ export function trimDescription(desc: string): string {
   }
   return desc;
 }
+
+export function flattenResources(
+  refs: ResourceRef[],
+  resources: { [s: string]: IKubeItem<IResource | IK8sList<IResource, {}>> },
+) {
+  const result: Array<IKubeItem<IResource | ISecret>> = [];
+  refs.forEach(ref => {
+    const kubeItem = resources[ref.getResourceURL()];
+    if (kubeItem) {
+      const itemList = kubeItem.item as IK8sList<IResource | ISecret, {}>;
+      if (itemList && itemList.items) {
+        itemList.items.forEach(i => {
+          result.push({
```

Comment:
So this won't handle a list of lists I don't think? But maybe that's never going to be a thing (don't know of a situation in k8s of such a list anyway).

Reply:
I don't know if a list of lists is allowed by the K8s API, certainly I've never seen one.